### PR TITLE
es6 (babel) support

### PR DESCRIPTION
**Pull request by jpka**

### CODE REVIEW

**Comment by jpka** [comment]
pretty self-explanatory, goes with the following:

https://github.com/noflo/noflo/pull/294
and
https://github.com/noflo/noflo-browser/pull/1

File: elements/noflo-component-editor.html [hunk 1]
```diff
@@ -304,6 +333,9 @@
         if (this.component.language === 'javascript') {
           return this.$.JavaScriptExample.textContent.trim();
         }
+        if (this.component.language === 'es6') {
```
Comment (subtleGradient) [inline elements/noflo-component-editor.html]: What about jsx? Also, ES6 and ES7 are called ES2015 now AFAIK.
Atom uses babel if it sees "use babel" in the code. Cf. http://blog.atom.io/2015/02/04/built-in-6to5.html

Cf. my comment here: https://github.com/noflo/noflo/pull/294#discussion-diff-38762552

File: elements/noflo-component-editor.html [hunk 2]
```diff
@@ -106,6 +106,35 @@
   return c;
 };
     </script>
+    <script type="text/plain" id="ES6Example">
```
Comment (subtleGradient) [inline elements/noflo-component-editor.html]: The convention is to use `<script type="text/babel"` or `<script type="text/ecmascript-6"`
Cf. https://babeljs.io/docs/usage/browser/

File: elements/noflo-new-component.html [hunk 3]
```diff
@@ -12,6 +12,7 @@
           <select name="type" value="{{ language }}">
             <option value="coffeescript">CoffeeScript</option>
             <option value="javascript">JavaScript</option>
+            <option value="es6">ES6 (Babel)</option>
```
Comment (subtleGradient) [inline elements/noflo-new-component.html]: It's "ES2015" now. Cf. https://babeljs.io/docs/learn-es2015/

**Pull request merged**